File: EssPortal.Models/Models/Request.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace EssPortal.Models
{
  

    [Table("Requests", Schema = "ess")]
    public class Request
    {
        [Key]  
        public int RequestID { get; set; } 
        public string RequsetPrivateNumber { get; set; } 
        [Required]
         public long EmployeeID { get; set; } 
        [Required] 
        public int RequestTypeID { get; set; }
        public virtual RequestType RequestType { get; set; }  
        [Required]
        public string StatusCode { get; set; } 
        public int UserID { get; set; }
        public virtual Users Users { get; set; } 
        public Nullable<bool> IsDelegate { get; set; }
        public Nullable<bool> IsDelegateApprove { get; set; }
        public Nullable<long> DelegateFromID { get; set; }
        public Nullable<long> DelegateToID { get; set; } 
        public Nullable<Boolean> IsReplacementEmployeeApprove { get; set; } 
    }

}

File: EssPortalAPI/ClientApp/src/app/views/orders/Models/Request.ts

class Requests {
    requestID: number;
    requsetPrivateNumber:string;
    employeeID: number;
    requestTypeID: number;
    statusCode: string; 
    userID: number;
    isDelegate: boolean;
    isDelegateApprove: boolean;
    delegateFromID: number;
    delegateToID: number;
    createdDate: Date;
    fileName: string;
}

class RequestExtraField {
    requestExtraFieldID: number;
    requestID: number;
    extraFieldTypeID:number ;
    value: string; 
}


class RequestType {
    requestTypeID: number;
    requestNameAr: string;
    requestNameEn: string;
    requestGroupID: number;
    isActive: boolean;
    icons: string;
}

class RequestStage {
    iD: number;
    stageTypeID: number;
    justification: string;
    actionCode: string;
    userID: number;
    createDate: string;
}

class Amendments {
    ID: number;
    amendmentReasonCode: number;
    requestID: number;
    type: string;
    time: string;
    fromDate: string;
    justification: string;
}

class Leave {
    ID: number;
    requestID: number;
    vacationTypeID: string;
    startDate: string;
    endDate: string;
    daysNo: number;
    replacmentStuffID: number;
    replacmentStuffIDIsApprove: boolean;
    justification: string;
}

class Housing {
    ID: number;
    requestID: number;
    startDate: string;
    endDate: string;
    TotalAmount: number;
    MonthlyInstallment: number;
    ContractPeriod: string;
    NoOfInstallment: boolean;
    LoanStartDate: string;
    LOANTYPECODE: string;
    justification: string;
}

class SearchRequestCritria{
    requestID: number;
    employeeID: number;
    managerID: number;
    requestTypeID: number;
    statusID: number;
}




export { Requests, RequestStage, Amendments, RequestType, Leave,
    RequestExtraField,
    SearchRequestCritria }